### ISSUE TITLE
gcs_list_objects rbind fails when contentType is missing from page of objects

### ISSUE THREAD

**Issue by camraynor** [cloudyr/googleCloudStorageR#104]
I've been working on a script that combines files stored in a large bucket and have been running into an error when `gcs_list_objects` pages through the bucket to list all of the objects (this seems to be somewhat similar to #73). The problem seems to occur when one of the pages doesn't have a column that is present in the other pages. In my case, this was the contentType column. 

When I run the command, I get the following error:
```
Error in rbind(deparse.level, ...) : 
  numbers of columns of arguments do not match
```
I printed the names of each item in the `req` dataframe before it is passed to `rbind` within the `gcs_list_objects` function using `trace`, and that gives the readout below. The error seems to occur when the contentType name is missing.
```
gcs_list_objects(bucket = "my-bucket-name", detail = "summary")
2019-07-01 04:20:15 -- Paging through results...
 [1] "id"                      "selfLink"                "name"                    "bucket"                 
 [5] "generation"              "metageneration"          "timeCreated"             "updated"                
 [9] "storageClass"            "timeStorageClassUpdated" "size"                    "md5Hash"                
[13] "mediaLink"               "crc32c"                  "etag"                    "contentType"            
[17] "componentCount"          "contentLanguage"        
2019-07-01 04:20:16 -- Paging through results...
 [1] "id"                      "selfLink"                "name"                    "bucket"                 
 [5] "generation"              "metageneration"          "contentType"             "timeCreated"            
 [9] "updated"                 "storageClass"            "timeStorageClassUpdated" "size"                   
[13] "md5Hash"                 "mediaLink"               "crc32c"                  "etag"                   
[17] "componentCount"          "contentLanguage"        
2019-07-01 04:20:16 -- Paging through results...
 [1] "id"                      "selfLink"                "name"                    "bucket"                 
 [5] "generation"              "metageneration"          "timeCreated"             "updated"                
 [9] "storageClass"            "timeStorageClassUpdated" "size"                    "md5Hash"                
[13] "mediaLink"               "crc32c"                  "etag"                    "componentCount"         
[17] "contentLanguage"        
Error in rbind(deparse.level, ...) : 
  numbers of columns of arguments do not match
```
The fix in my case was to redefine the function using `dyplr`'s `bind_rows` function in place of the `rbind`. That solved the problem in my code, although I know `dplyr` isn't required by `googleCloudStorageR`. I expect that adding some code to add a corresponding column of `NA`'s when a column is missing from either the `req` or `more_req` dataframes would also do the trick.

**Comment by MarkEdmondson1234** [comment]
Would you mind showing which version you are using (sessionInfo() ) as this was an issue that I thought was solved in an earlier version, so checking if it only needs an update.


________________________________

**Comment by camraynor** [comment]
No problem. I'm using version 0.4.0:
```R version 3.6.0 (2019-04-26)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Debian GNU/Linux 9 (stretch)

Matrix products: default
BLAS/LAPACK: /usr/lib/libopenblasp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8       
 [4] LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
[10] LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] furrr_0.1.0                future_1.13.0              googleCloudStorageR_0.4.0 
 [4] googleComputeEngineR_0.3.0 forcats_0.4.0              stringr_1.4.0             
 [7] dplyr_0.8.1                purrr_0.3.2                readr_1.3.1               
[10] tidyr_0.8.3                tibble_2.1.3               ggplot2_3.1.1             
[13] tidyverse_1.2.1           

loaded via a namespace (and not attached):
 [1] tidyselect_0.2.5  listenv_0.7.0     haven_2.1.0       lattice_0.20-38   vctrs_0.1.0      
 [6] colorspace_1.4-1  generics_0.0.2    yaml_2.2.0        utf8_1.1.4        rlang_0.3.4      
[11] pillar_1.4.1      glue_1.3.1        withr_2.1.2       modelr_0.1.4      readxl_1.3.1     
[16] plyr_1.8.4        munsell_0.5.0     gtable_0.3.0      cellranger_1.1.0  zip_2.0.2        
[21] rvest_0.3.4       codetools_0.2-16  memoise_1.1.0     parallel_3.6.0    curl_3.3         
[26] fansi_0.4.0       broom_0.5.2       Rcpp_1.0.1        openssl_1.4       scales_1.0.0     
[31] backports_1.1.4   jsonlite_1.6      googleAuthR_0.7.0 hms_0.4.2         askpass_1.1      
[36] digest_0.6.19     stringi_1.4.3     grid_3.6.0        cli_1.1.0         tools_3.6.0      
[41] magrittr_1.5      lazyeval_0.2.2    zeallot_0.1.0     crayon_1.3.4      pkgconfig_2.0.2  
[46] xml2_1.2.0        lubridate_1.7.4   assertthat_0.2.1  httr_1.4.0        rstudioapi_0.10  
[51] R6_2.4.0          globals_0.12.4    nlme_3.1-139      compiler_3.6.0  
```

**Comment by MarkEdmondson1234** [comment]
Cool that should be then covered by the fix in the dev version that relies on googleAuthR version 0.8.0 that was just published - if you install CRAN googleAuthR 0.8.0 and github version googleCloudStorageR this should be fixed (I install via remotes::install_github(“cloudy/googleCloudStorageR”) )


________________________________

**Comment by camraynor** [comment]
Excellent—thank you! That seems to have solved it.
